nifi-nar-bundles/nifi-windows-event-log-bundle/nifi-windows-event-log-processors/src/main/java/org/apache/nifi/processors/windows/event/log/ConsumeWindowsEventLog.java

```diff
@@ -0,0 +1,282 @@
+import org.apache.nifi.processors.windows.event.log.jna.WEvtApi;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+
+@InputRequirement(InputRequirement.Requirement.INPUT_FORBIDDEN)
+@Tags({"ingest", "event", "windows"})
+@TriggerSerially
+@CapabilityDescription("Registers a Windows Event Log Subscribe Callback to receive FlowFiles from Events on Windows.  These can be filtered via channel and XPath.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "mime.type", description = "Will set a MIME type value of application/xml.")
+})
+public class ConsumeWindowsEventLog extends AbstractSessionFactoryProcessor {
+    public static final String DEFAULT_CHANNEL = "System";
+    public static final String DEFAULT_XPATH = "*";
+    public static final int DEFAULT_MAX_BUFFER = 1024 * 1024;
+    public static final int DEFAULT_MAX_QUEUE_SIZE = 1024;
+
+    public static final PropertyDescriptor CHANNEL = new PropertyDescriptor.Builder()
+            .name("channel")
+            .displayName("Channel")
+            .required(true)
+            .defaultValue(DEFAULT_CHANNEL)
+            .description("The Windows Event Log Channel to listen to.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("query")
+            .displayName("XPath Query")
+            .required(true)
+            .defaultValue(DEFAULT_XPATH)
+            .description("XPath Query to filter events. (See https://msdn.microsoft.com/en-us/library/windows/desktop/dd996910(v=vs.85).aspx for examples.)")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MAX_BUFFER_SIZE = new PropertyDescriptor.Builder()
+            .name("maxBuffer")
+            .displayName("Maximum Buffer Size")
+            .required(true)
+            .defaultValue(Integer.toString(DEFAULT_MAX_BUFFER))
+            .description("The individual Event Log XMLs are rendered to a buffer." +
+                    "  This specifies the maximum size in bytes that the buffer will be allowed to grow to. (Limiting the maximum size of an individual Event XML.)")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MAX_EVENT_QUEUE_SIZE = new PropertyDescriptor.Builder()
+            .name("maxQueue")
+            .displayName("Maximum queue size")
+            .required(true)
+            .defaultValue(Integer.toString(DEFAULT_MAX_QUEUE_SIZE))
+            .description("Maximum number of events to queue for transformation into FlowFiles.")
```

Comment:
This should offer more explanation for users who have no idea about the internals of the processor (they are unaware that there is an internal queue of events waiting to be turned into flowfiles on each onTrigger).